pkg/mountutil/mountutil.go

```diff
@@ -54,77 +46,17 @@ func ProcessFlagV(s string, volStore volumestore.VolumeStore) (*Processed, error
 		res      Processed
 		src, dst string
 		options  []string
+		err      error
 	)
 
-	split := strings.Split(s, ":")
-	switch len(split) {
-	case 1:
-		dst = s
-		res.AnonymousVolume = idgen.GenerateID()
-		logrus.Debugf("creating anonymous volume %q, for %q", res.AnonymousVolume, s)
-		anonVol, err := volStore.Create(res.AnonymousVolume, []string{})
-		if err != nil {
-			return nil, err
-		}
-		src = anonVol.Mountpoint
-		res.Type = Volume
-	case 2, 3:
-		res.Type = Bind
-		src, dst = split[0], split[1]
-		if !strings.Contains(src, "/") {
-			// assume src is a volume name
-			res.Name = src
-			vol, err := volStore.Get(src)
-			if err != nil {
-				if errors.Is(err, errdefs.ErrNotFound) {
-					vol, err = volStore.Create(src, nil)
-					if err != nil {
-						return nil, err
-					}
-				} else {
-					return nil, err
-				}
-			}
-			// src is now full path
-			src = vol.Mountpoint
-			res.Type = Volume
-		}
-		if !filepath.IsAbs(src) {
-			logrus.Warnf("expected an absolute path, got a relative path %q (allowed for nerdctl, but disallowed for Docker, so unrecommended)", src)
-			var err error
-			src, err = filepath.Abs(src)
-			if err != nil {
-				return nil, err
-			}
-		}
-		if !filepath.IsAbs(dst) {
-			return nil, fmt.Errorf("expected an absolute path, got %q", dst)
-		}
-		rawOpts := ""
-		if len(split) == 3 {
-			rawOpts = split[2]
-		}
-		res.Mode = rawOpts
-
-		// always call parseVolumeOptions for bind mount to allow the parser to add some default options
-		var err error
-		var specOpts []oci.SpecOpts
-		options, specOpts, err = parseVolumeOptions(res.Type, src, rawOpts)
-		if err != nil {
-			return nil, err
-		}
-		res.Opts = append(res.Opts, specOpts...)
-	default:
-		return nil, fmt.Errorf("failed to parse %q", s)
+	src, dst, options, err = ProcessSplit(s, volStore, &res, src, dst, options)
+	if err != nil {
+		return nil, err
 	}
-
-	fstype := "nullfs"
 	if runtime.GOOS != "freebsd" {
-		fstype = "none"
 		options = append(options, "rbind")
 	}
 	res.Mount = specs.Mount{
-		Type:        fstype,
```

Comment:
Why is this being dropped?  I don't see it moved anywhere.

Reply:
Because of this the mounted type during inspect was null. Now I get this output: 

```
D:\GolandProjects\github.com\nerdctl\cmd\nerdctl>nerdctl inspect 3c46c12ed1f1
[
    {
        "Id": "3c46c12ed1f1f6272b4ab3f93996a19e2e52667313c2a2cbbf94f87b08350dad",
        "Created": "2022-05-06T09:54:49.1905099Z",
        "Path": "c:\\windows\\system32\\cmd.exe",
        "Args": null,
        "State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "Pid": 10928,
            "ExitCode": 0,
            "FinishedAt": "2022-05-06T09:54:51.2735242Z"
        },
        "Image": "mcr.microsoft.com/windows/nanoserver:20H2",
        "ResolvConfPath": "",
        "HostnamePath": "",
        "LogPath": "",
        "Name": "nanoserver-3c46c",
        "RestartCount": 0,
        "Driver": "windows",
        "Platform": "windows",
        "AppArmorProfile": "",
        "Mounts": [
            {
                "Type": "volume",
                "Name": "500907e8f1465b04f132f5a49b76c59c1e41f18aa9063faffa5c2bb7fc2e72a5",
                "Source": "C:\\ProgramData\\nerdctl\\052055e3\\volumes\\default\\500907e8f1465b04f132f5a49b76c59c1e41f18aa9063faffa5c2bb7fc2e72a5\\_data",
                "Destination": "C:\\src",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            }
        ],
        "NetworkSettings": {
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "MacAddress": "",
            "Networks": {}
        }
    }
]
```
Please check.

Reply:
is only printing `null` an issue on Windows?  I am concerned this change affects something on freebsd or linux.  The tests aren't failing but maybe we are missing coverage here

Reply:
> If we pass "none" as fstype for windows, it gives this error: "time="2022-05-12T12:13:13+05:30" level=fatal msg="invalid OCI spec - Type 'none' not supported: unknown"".


> why are we passing nullfs or none, when we are already passing mount types like bind, volume or tmps while splitting the source and destination.

I don't know why none/nullfs is passed here.  @AkihiroSuda do you have any insight?